pkg/util/liveconfig/hive.go

```diff
@@ -21,14 +23,52 @@ const (
 	hiveAdoptEnableEnvVar     = "ARO_ADOPT_BY_HIVE"
 )
 
-func parseKubeconfig(credentials []mgmtcontainerservice.CredentialResult) (*rest.Config, error) {
-	res := make([]byte, base64.StdEncoding.DecodedLen(len(*credentials[0].Value)))
-	_, err := base64.StdEncoding.Decode(res, *credentials[0].Value)
+func getAksKubeconfig(ctx context.Context, managedClustersClient containerservice.ManagedClustersClient, location string, index int, credentialType AksCredentialType) (*rest.Config, error) {
+	aksClusterName := fmt.Sprintf("aro-aks-cluster-%03d", index)
+
+	aksClusters, err := managedClustersClient.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var aksCluster *mgmtcontainerservice.ManagedCluster
+
+outerLoop:
+	for aksClusters.NotDone() {
+		for _, cluster := range aksClusters.Values() {
+			if *cluster.Name == aksClusterName && *cluster.Location == location {
+				aksCluster = &cluster
+				break outerLoop
+			}
+		}
+		err = aksClusters.NextWithContext(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if aksCluster == nil {
+		return nil, fmt.Errorf("failed to find the AKS cluster %s in %s", aksClusterName, location)
+	}
+
+	aksResourceGroup := strings.Replace(*aksCluster.NodeResourceGroup, fmt.Sprintf("-aks%d", index), "", 1)
+
+	var res mgmtcontainerservice.CredentialResults
+	if credentialType == AdminCredentials {
+		res, err = managedClustersClient.ListClusterAdminCredentials(ctx, aksResourceGroup, aksClusterName, "public")
+	} else {
+		res, err = managedClustersClient.ListClusterUserCredentials(ctx, aksResourceGroup, aksClusterName, "public")
+	}
+
 	if err != nil {
 		return nil, err
 	}
 
-	clientconfig, err := clientcmd.NewClientConfigFromBytes(res)
+	return parseKubeconfig(*res.Kubeconfigs)
+}
+
+func parseKubeconfig(credentials []mgmtcontainerservice.CredentialResult) (*rest.Config, error) {
+	clientconfig, err := clientcmd.NewClientConfigFromBytes(*credentials[0].Value)
```

Comment:
According to this doc, the `value` is base64 encoded, does `NewCLientConfigFromBytes` account for this?
https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters/list-cluster-user-credentials?tabs=Go#credentialresult